docs/madr/decisions/056-meshservice-migration.md

```diff
@@ -0,0 +1,180 @@
+# MeshService migration
+
+* Status: accepted
+
+## Context and Problem Statement
+
+This MADR addresses the switch from not having MeshServices at all to MeshServices
+being automatically created, either from `Services` in Kubernetes or from
+`Dataplanes` in Universal.
+
+When upgrading from 2.8 to 2.9, nothing at the data plane level should change.
+But MeshService behaves differently in several different ways.
+
+In general the migration process for MeshService consists of no longer using
+`kuma.io/service` so:
+
+- targeting policies to the real `MeshService` resource
+- switching over to the `MeshService` i.e. `HostnameGenerator` hostnames, if not
+  using Kubernetes DNS
+
+### Policy
```

Comment:
I have a question about policy API, are we going to deprecate and later remove `MeshService` from `spec.targetRef`? I am asking because we will most likely move Mesh*Routes to `spec.to[].targetRef` and since `MeshService` is an outbound, it does not make sense to put it in `spec.targetRef`

Reply:
Yes that's the goal. I don't think it affects this MADR though, it's basically just a rule: "wherever kuma.io/service is targeted, also target MeshService generated for that kuma.io/service"